membership-attribute-service/app/services/SupporterRatePlanToAttributesMapper.scala

```diff
@@ -0,0 +1,321 @@
+package services
+
+import models.{Attributes, SupporterRatePlanItem}
+import org.joda.time.LocalDate
+import services.SupporterRatePlanToAttributesMapper.productRatePlanMappings
+
+class SupporterRatePlanToAttributesMapper(stage: String) {
+
+  def attributesFromSupporterRatePlans(identityId: String, supporterRatePlanItems: List[SupporterRatePlanItem]) = {
+    supporterRatePlanItems.foldLeft[Option[Attributes]](None) { (maybeAttributes, item) =>
+      mapRatePlanToAttributes(maybeAttributes, item, identityId)
+    }
+  }
+
+  private def mapRatePlanToAttributes(maybeAttributes: Option[Attributes], ratePlanItem: SupporterRatePlanItem, identityId: String) =
+    productRatePlanMappings(stage)
+      .collectFirst {
+        case (ids, transformFunction) if ids.contains(ratePlanItem.productRatePlanId) =>
+          Some(transformFunction(maybeAttributes.getOrElse(Attributes(identityId)), ratePlanItem))
+      }
+      .getOrElse(maybeAttributes)
+
+}
+
+object SupporterRatePlanToAttributesMapper {
+
+  val productRatePlanMappings: Map[String, Map[List[String], (Attributes, SupporterRatePlanItem) => Attributes]] =
```

Comment:
this type is a little hard to digest - might be worth making the `(Attributes, SupporterRatePlanItem) => Attributes` part of the type into a single abstract method trait, and possibly adding type aliases for Stage and ProductRatePlanId